cloudant-client/src/test/java/com/cloudant/tests/CloudantClientHelper.java

```diff
@@ -118,7 +121,8 @@ public static ClientBuilder newMockWebServerClientBuilder(MockWebServer mockServ
     public static ClientBuilder getClientBuilder() {
         return ClientBuilder.url(SERVER_URL)
                 .username(COUCH_USERNAME)
-                .password(COUCH_PASSWORD);
+                .password(COUCH_PASSWORD)
+                .iamApiKey(COUCH_IAM_API_KEY);
```

Comment:
It may be worth adding a comment here that this test helper is relying on the behaviour of the `ClientBuilder` to do the right thing. I wonder if we should make this return a different builder in each case, and/or add some additional tests to validate that the behaviour remains as expected when combinations are supplied (since we're not throwing exceptions if someone specifies both username/pw and an IAM API key).

e.g.
```java
ClientBuilder builder = ClientBuilder.url(SERVER_URL);
if (COUCH_IAM_API_KEY != null) {
  return builder..iamApiKey(COUCH_IAM_API_KEY);
} else {
  return builder.username(COUCH_USERNAME)
    .password(COUCH_PASSWORD);
}
```
and tests for `testIAMApiKeyPreferredToUsernamePassword()` `testUsernamePasswordFallbackIfNoAPIKey()` or similar.

I think as it stands there is scope for the `build()` behaviour to change and we wouldn't realise that we were testing with the wrong creds.

We should probably also document the behaviour of specifying credentials and a key if we didn't already.

Reply:
@ricellis my preference would be that this helper remains as-is and we have a test like `testIAMApiKeyPreferredToUsernamePassword` as suggested